docker-compose.yaml

```diff
@@ -0,0 +1,306 @@
+    environment:
+      - GOMAXPROCS=2
+      - 'SRC_FRONTEND_INTERNAL=sourcegraph-frontend-internal:3090'
+      - JAEGER_AGENT_HOST=jaeger-agent
+    volumes:
+      - '~/sourcegraph-docker/searcher-0-disk:/mnt/cache'
+    networks:
+      - sourcegraph
+    restart: always
+  syntect-server:
+    container_name: syntect-server
+    image: >-
+      sourcegraph/syntect_server:2b5a3fb@sha256:ef5529cafdc68d5a21edea472ee8ad966878b173044aa5c3db93bc3d84765b1f
+    networks:
+      - sourcegraph
+    restart: always
+  symbols-0:
+    container_name: symbols-0
+    image: >-
+      sourcegraph/symbols:3.10.4@sha256:7ef128b652fe554de4ae38bb4ee592568b315c84b122a577980042d89cd6e40f
+    environment:
+      - GOMAXPROCS=2
+      - 'SRC_FRONTEND_INTERNAL=sourcegraph-frontend-internal:3090'
+      - JAEGER_AGENT_HOST=jaeger-agent
+    volumes:
+      - '~/sourcegraph-docker/symbols-0-disk:/mnt/cache'
+    networks:
+      - sourcegraph
+    restart: always
+  zoekt-indexserver-0:
+    container_name: zoekt-indexserver-0
+    image: >-
+      sourcegraph/zoekt-indexserver:0.0.20191031121923-5bd7e84@sha256:43eac5fddcc32f84f097e1466918ec9ac2fe54fe34649b8399de93963d9e4916
+    environment:
+      - GOMAXPROCS=8
+      - 'HOSTNAME=zoekt-webserver-0:6070'
+      - 'SRC_FRONTEND_INTERNAL=http://sourcegraph-frontend-internal:3090'
+    volumes:
+      - '~/sourcegraph-docker/zoekt-0-shared-disk:/data/index'
+    networks:
+      - sourcegraph
+    restart: always
+    hostname: zoekt-indexserver-0
+  zoekt-webserver-0:
+    container_name: zoekt-webserver-0
+    image: >-
+      sourcegraph/zoekt-webserver:0.0.20191031121751-5bd7e84@sha256:1afbfb746a3d43a7532693093d2f993dac5b5536037b31d3ab305ceef5687380
+    environment:
+      - GOMAXPROCS=16
+      - 'HOSTNAME=zoekt-webserver-0:6070'
+    volumes:
+      - '~/sourcegraph-docker/zoekt-0-shared-disk:/data/index'
+    networks:
+      - sourcegraph
+    restart: always
+    hostname: zoekt-webserver-0
+  sourcegraph-frontend-internal:
+    container_name: sourcegraph-frontend-internal
+    image: >-
+      sourcegraph/frontend:3.10.4@sha256:5791fb16a1a2828f14c227160104eba3a251c876359e0cb33c48f832e3d2a53f
+    environment:
+      - GOMAXPROCS=4
+      - PGHOST=pgsql
+      - 'SRC_GIT_SERVERS=gitserver-0:3178'
+      - 'SRC_SYNTECT_SERVER=http://syntect-server:9238'
+      - 'SEARCHER_URL=http://searcher-0:3181'
+      - 'SYMBOLS_URL=http://symbols-0:3184'
+      - 'INDEXED_SEARCH_SERVERS=zoekt-webserver-0:6070'
+      - 'SRC_FRONTEND_INTERNAL=sourcegraph-frontend-internal:3090'
+      - 'REPO_UPDATER_URL=http://repo-updater:3182'
+      - 'REPLACER_URL=http://replacer:3185'
+      - 'LSIF_SERVER_URL=http://lsif-server:3186'
+      - 'GRAFANA_SERVER_URL=http://grafana:3000'
+    volumes:
+      - >-
+        ~/sourcegraph-docker/sourcegraph-frontend-internal-0-disk:/mnt/cache
+    networks:
+      - sourcegraph
+    restart: always
+  sourcegraph-frontend-2:
```

Comment:
I assume that we'll want to get rid of the multiple frontend containers from the default version of this config?

Reply:
We should investigate if we can use https://docs.docker.com/compose/compose-file/#replicas for this

Reply:
@slimsag That feature is only supported for `docker swarm`, and it seems really unclear what the future of swarm is: https://github.com/docker/swarmkit/issues/2665

Regardless, I don't have personal experience with it. I can explore it more, but I'm hesitant to recommend it since its status seems to be in limbo.